### PR TITLE
Add egress-masquerade-interfaces option to cilium

### PR DESCRIPTION
**Pull request by ahreehong**
*Issue #, if available:*

*Description of changes:*
Adds support for configuring the egress-masquerade-interfaces option in Cilium CNI via eks-a cluster spec. This allows users to configure this setting at cluster create as well as upgrade the setting.

Users can configure this setting like so:

```yaml
  clusterNetwork:
    cniConfig:
      cilium:
        egressMasqueradeInterfaces: eth+
```
where `egressMasqueradeInterfaces` expects a string that is an interface name or an interface prefix.

For more context on `egressMasqueradeInterfaces`:
https://docs.cilium.io/en/v1.13/network/concepts/masquerading/#iptables-based

*Testing (if applicable):*
Checked the cilium config map to validate that the setting had been configured correctly, and validated that the setting could be changed via upgrades.

Also validated that setting this resulted in the correct behavior via:
1. Setting up hardware with two network interfaces.
2. Creating a cluster with one interface specified in `egressMasqueradeInterfaces`
3. Checking that the correct interface was specified on the cilium config map
4. `tcp dump` on the non-specified interface and validating that the IP that gets used when pinging `1.1.1.1` is non-masqueraded

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jacobweinstock — CHANGES_REQUESTED** [review]

File: pkg/networking/cilium/upgrader_test.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ func TestUpgraderUpgradeSuccessValuesChanged(t *testing.T) {
 
 	// setting policy enforcement mode to something other than the "default" mode
 	tt.newSpec.Cluster.Spec.ClusterNetwork.CNIConfig.Cilium.PolicyEnforcementMode = v1alpha1.CiliumPolicyModeNever
+	// tt.newSpec.Cluster.Spec.ClusterNetwork.CNIConfig.Cilium.EgressMasqueradeInterfaces = "test"
```
[the remark this answers is not in the document]
Comment (ahreehong) [reply pkg/networking/cilium/upgrader_test.go:111]: Yeah, I missed this when I was cleaning up the code. Will remove

File: config/manifest/eksa-components.yaml [hunk 2]
```diff
@@ -6542,7 +6547,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.uid
-        image: public.ecr.aws/l0g8r8j6/eks-anywhere-cluster-controller:v0.0.0-eks-a-v0.0.0-dev-build.251
+        image: public.ecr.aws/l0g8r8j6/eks-anywhere-cluster-controller:v0.16.0-eks-a-v0.0.0-dev-build.7080
```
[the remark this answers is not in the document]
Comment (ahreehong) [reply config/manifest/eksa-components.yaml:6550]: Yeah this is left over from local testing. I'm not sure if this value should be updated though, `public.ecr.aws/l0g8r8j6/eks-anywhere-cluster-controller:v0.0.0-eks-a-v0.0.0-dev-build.251` seems like an ancient build

File: config/crd/bases/anywhere.eks.amazonaws.com_clusters.yaml [hunk 3]
```diff
@@ -66,6 +66,11 @@ spec:
                     properties:
                       cilium:
                         properties:
+                          egressMasqueradeInterfaces:
+                            description: EgressMasquaradeInterfaces determines which
+                              network interfaces are used for masquerading. Accepted
+                              values are strings.
```
[the remark this answers is not in the document]
Comment (ahreehong) [reply config/crd/bases/anywhere.eks.amazonaws.com_clusters.yaml]: test

**Review by jacobweinstock — APPROVED** [review]
thanks for the patience and work on this! @ahreehong

**Pull request merged**